test/WebJobs.Script.Tests/Controllers/Admin/FunctionsControllerTests.cs

```diff
@@ -85,6 +94,46 @@ public async Task Invoke_CallsFunction()
             Assert.True(functionInvoked);
         }
 
+        [Theory]
+        [InlineData(false, HttpStatusCode.Created)]
+        [InlineData(true, HttpStatusCode.BadRequest)]
+        public async Task CreateOrUpdate_DependsOnReadOnlyFileSystem(bool readOnlyFileSystem, HttpStatusCode expected)
+        {
+            var scriptPath = Path.GetTempPath();
+            var applicationHostOptions = new ScriptApplicationHostOptions
+            {
+                ScriptPath = scriptPath,
+                IsFileSystemReadOnly = readOnlyFileSystem
+            };
+
+            string testFunctionName = "TestFunction";
+            var functionsMetadataMock = new Mock<FunctionMetadataResponse>();
+            var functionsManagerMock = new Mock<IWebFunctionsManager>();
+            functionsManagerMock.Setup(p => p.CreateOrUpdate(It.IsAny<string>(), It.IsAny<FunctionMetadataResponse>(), It.IsAny<HttpRequest>()))
+            .Callback<string, FunctionMetadataResponse, HttpRequest>((name, functionMetadata, request) =>
+            {
+                // verify the correct arguments were passed to the method
+                Assert.Equal(testFunctionName, name);
+                Assert.Equal(request.Method, "Put");
+            })
+            .Returns(Task.FromResult((true, true, functionsMetadataMock.Object)));
+
+            var mockRouter = new Mock<IWebJobsRouter>();
+            var optionsWrapper = new OptionsWrapper<ScriptApplicationHostOptions>(applicationHostOptions);
+            var testController = new FunctionsController(functionsManagerMock.Object, mockRouter.Object, new LoggerFactory(), optionsWrapper);
+
+            var httpContext = new DefaultHttpContext();
+            httpContext.Request.Host = new HostString("local");
+            httpContext.Request.Method = "Put";
+            httpContext.Request.Path = $"/admin/functions/{testFunctionName}";
+            testController.ControllerContext.HttpContext = httpContext;
+
+            var fileMonitoringServiceMock = new Mock<IFileMonitoringService>();
+            var result = (ObjectResult)(await testController.CreateOrUpdate(testFunctionName, functionsMetadataMock.Object, fileMonitoringServiceMock.Object));
+
+            Assert.Equal((int)expected, result.StatusCode);
```

Comment:
assert error message as well. Also, extend the test to include other cases that can cause 400 - empty function name?

Reply:
thanks, updated!